### PR TITLE
Add date and orgUnitGroup filters to Organization Unit Sync

### LINKED ISSUE #44: OU Filter & Sync Enhancements
- [ ] Integrate filters: Org Unit Group Filter and Date Filter.

For later:
- [ ] Send internal messaging on init, complete and failure
- [ ] Evaluate OU tree integration 
- [ ] Document

### PR DESCRIPTION
**Pull request by matiasgarcia91**
### :pushpin: References

* **Issue:** Closes #44 

### :memo: Implementation
Modified d2-ui-components `DatePicker` and created `Dropdown`. Modified `BaseSyncConfigurator` table to accept extra filters from parents.

### :art: Screenshots
image
image

### :fire: Is there anything the reviewer should know to test it?
d2-ui-components library still hasn't been updated so the datePicker will look bad, but it works anyway.

### :bookmark_tabs: Others

- Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?
https://github.com/EyeSeeTea/d2-ui-components/pull/24

### CODE REVIEW

File: src/components/sync-configurator/OrganisationUnitSync.jsx [hunk 1]
```diff
@@ -3,16 +3,58 @@ import BaseSyncConfigurator from "./BaseSyncConfigurator";
...
 import { OrganisationUnitModel } from "../../models/d2Model";
+import Dropdown from "../shared/Dropdown";
 
 export default class OrganisationUnitSync extends React.Component {
     static propTypes = {
         d2: PropTypes.object.isRequired,
     };
 
+    state = {
+        dropdown: {
```
Comment (tokland) [inline src/components/sync-configurator/OrganisationUnitSync.jsx]: Name of property should reflect the what (orgUnitGroup?) not the how (that is a dropdown element)

File: src/models/d2Model.ts [hunk 2]
```diff
@@ -18,13 +18,17 @@ abstract class D2Model {
...
+        const { search = null, date = null, d2Filters = [] } = filters || {};
         const { page = 1, pageSize = 20, sorting = this.initialSorting } = pagination || {};
         const fields = this.details.map(e => e.name);
 
         const [field, direction] = sorting;
         const order = `${field}:i${direction}`;
-        const filter = _.compact([search ? `displayName:ilike:${search}` : null]);
+        const filter = _.compact([
+            search ? `displayName:ilike:${search}` : null,
+            date ? `lastUpdated:gt:${date.toISOString()}` : null,
```
Comment (tokland) [inline src/models/d2Model.ts]: Minor: Maybe ge?

File: src/components/sync-configurator/BaseSyncConfigurator.jsx [hunk 3]
```diff
@@ -30,12 +40,31 @@ class BaseSyncConfigurator extends React.Component {
...
+        this.setState({ filters: { date: value } });
+    };
 
+    renderCustomFilters = () => {
+        const { renderExtraFilters } = this.props;
+        const { date } = this.state.filters;
+        return (
+            <React.Fragment>
+                <DatePicker
+                    placeholder={"Enter date"}
```
[the remark this answers is not in the document]
Comment (SferaDev) [reply src/components/sync-configurator/BaseSyncConfigurator.jsx]: DatePicker has label as required.

Date is lastUpdated

[the remark this answers is not in the document]
Comment (matiasgarcia91) [reply src/components/sync-configurator/BaseSyncConfigurator.jsx]: What do you think would be a good literal for this? "Last updated date" I can't think of anything concise that transmits "From selected date onwards".

@SferaDev that prop is changed to optional on d2-ui-components PR

[the remark this answers is not in the document]
Comment (matiasgarcia91) [reply src/components/sync-configurator/BaseSyncConfigurator.jsx]: good catch

File: src/components/sync-configurator/OrganisationUnitSync.jsx [hunk 4]
```diff
@@ -3,16 +3,58 @@ import BaseSyncConfigurator from "./BaseSyncConfigurator";
...
+
+    handleFilterChange = event => {
+        const { items } = this.state.dropdown;
+        this.setState({ dropdown: { value: event.target.value, items } });
+    };
+
+    renderExtraFilters = () => {
+        const { items, value } = this.state.dropdown;
+        return (
+            <Dropdown
```
[the remark this answers is not in the document]
Comment (matiasgarcia91) [reply src/components/sync-configurator/OrganisationUnitSync.jsx]: What do you think is best for this? For now I permitted the placeholder to be selectable and changed it to italics to give a _default_ look. I've been avoiding labels to maintain the aesthetic of the form but maybe i should give up on that and use one.

**Pull request merged**